packages/connect/src/device/Device.ts

```diff
@@ -567,6 +574,10 @@ export class Device extends TypedEmitter<DeviceEvents> {
         const { message } = await this.getCommands().typedCall('GetFeatures', 'Features', {});
         this._updateFeatures(message);
 
+        if (this.authenticityChecks.firmwareHash === null) {
+            this.authenticityChecks.firmwareHash = await this.checkFirmwareHash();
+        }
+
```

Comment:
not a good place, it would get too many invocations. not sure what is the best place but for example `initialize` is surely better